nilearn/maskers/surface_labels_masker.py

```diff
@@ -468,17 +473,23 @@ def inverse_transform(self, signals):
 
         return imgs
 
-    def _reporting(self):
+    def _reporting(self) -> list:
         """Load displays needed for report.
 
         Returns
         -------
         displays : list
             A list of all displays to be rendered.
         """
+        # Handle the edge case where this function is called
+        # without matplolib or
+        # with a masker having report capabilities disabled
+        if not is_matplotlib_installed() or not self._has_report_data():
+            return [None]
+
```

Comment:
Could this be done for all masker `reporting` functions and remove `not is_matplotlib_installed()` from `create_figure_for_report` function?